Add ability to specify colors/styles in `status` command

knelli2 opened the pull request:
Currently this can only change the style of 'RUNNING', 'FAILED', etc. but can be extended in the future

## Proposed changes

Since everybody uses different colors on their terminal, the defaults may not show up properly. This allows the user to specify in the config file color/styles that will override the defaults so they can choose whatever colors they want (not all options have to be specified). The config file should look like

```yaml
status:
  state_styles:
    RUNNING: [bright_green]
    FAILED: [grey]
```

<!--
At a high level, describe what this PR does.
-->

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsvu commented:
@knelli2 I added a general config file in #4916, so all you need to do here is add an option like `--state-styles`. The configuration from the file will be passed to it automatically. You can note in the help string that the option should be specified in the config file, then you don't need to worry about parsing a dictionary from the command line.

knelli2 commented:
Should the default to this new option just be `None`?

nilsvu commented:
Empty dict I think. Edit: Actually it's up to you how to handle that case. Empty dict probably works well, but you can also handle `None`.

knelli2 committed:
Allow specifying colors/styles for status command

These can either be passed as a dict to the command line or
specified in the spectre config file

knelli2 commented on tools/Status/Status.py:
```diff
@@ -340,6 +349,15 @@ def render_status(show_paths, show_unidentified, **kwargs):
               default=None,
               help=("On a new screen, refresh jobs every 'watch' number "
                     "of seconds. Exit out with Ctl+C."))
+@click.option("--state-styles",
+              "input_state_styles",
```
@nilsvu When trying this on wheeler, I had to specify `input_state_styles` in the config file rather than just `state_styles`. Do you know why that is?

It's not a big deal, I can just swap around the names, but was curious why it chose the variable name and not the option name.

nilsvu replied:
Click uses the variable name as key in the `default_map` that it passes to the command. Just delete the `input_state_styles` name and pass it through as `state_styles` (see comment above).

nilsvu commented on tools/Status/Status.py:
```diff
@@ -154,16 +154,9 @@ def _state_order(state):
-            "TIMEOUT": "[red]",
-        }
-        return style.get(value, "") + str(value)
+        return state_styles.get(value, "") + str(value)
```
do this:

```py
style = {
  # ... (same as before)
}
style.update(state_styles)
```

Then you don't have to change any other code, only pass the `state_styles` from the options through.

nilsvu commented on tools/Status/Status.py:
```diff
@@ -340,6 +349,15 @@ def render_status(show_paths, show_unidentified, **kwargs):
+@click.option("--state-styles",
+              "input_state_styles",
+              type=dict,
+              default=None,
```
If you make this an empty dict then it's easier to handle (`dict.update` does nothing).

knelli2 commented:
Ok yup those changes made everything a lot simpler

nilsvu commented on tools/Status/Status.py:
```diff
@@ -340,6 +343,14 @@ def render_status(show_paths, show_unidentified, **kwargs):
+              help=("Dictionary between sacct states and rich modifiers for "
+                    "how the state will be printed. Rather than always having "
+                    "to specify a dict on the command line, you can add this "
+                    "to the spectre config file. See `spectre -h` for its "
```
Could you give an example here?
